### PR TITLE
Added arc area

### PR DESCRIPTION
**Pull request by mithru**
#33 Area of arc

### CODE REVIEW

File: src/entities/shapeEntity.js [hunk 1]
```diff
@@ -23,9 +23,31 @@ function ShapeEntity(Interceptor,shapeObject,arguments, canvasX, canvasY) {
...
-    var objectArea = 0;
+    let objectArea = 0;
     if (!objectType.localeCompare('arc')) {
-      objectArea = 0;
+      // area of full ellipse = PI * horizontal radius * vertical radius.
+      // therefore, area of arc = difference bet. arc's start and end radians * horizontal radius * vertical radius.
+      // the below expression is adjusted for negative values and differences in arc's start and end radians over PI*2  
+      var arcSizeInRadians = ((((arguments[5] - arguments[4]) % (PI * 2)) + (PI * 2)) % (PI * 2)); 
+      objectArea = arcSizeInRadians * arguments[2] * arguments[3] / 8;
+      if(arguments[6] === 'open' || arguments[6] === 'chord' ){
```
Comment (mithru) [inline src/entities/shapeEntity.js]: any idea why it's going lower-case? I initially tried CHORD and OPEN but did not work. Is the library reading all arguments and converting to lowercase somewhere?

Comment (MathuraMG) [reply src/entities/shapeEntity.js]: did you check what it passes as in the arguments? Maybe that is lower case?

File: dist/p5-accessibility.js [hunk 2]
```diff
@@ -1292,9 +1291,31 @@ Registry.register(FillEntity);
...
+        let Cx = arguments[0] + (arguments[2]/2) * cos(arguments[5]).toFixed(2);
+        let Cy = arguments[1] + (arguments[3]/2) * sin(arguments[5]).toFixed(2);
+        let areaOfExtraTriangle = abs(Ax*(By - Cy) + Bx*(Cy - Ay) + Cx*(Ay - By))/2;
+        if(arcSizeInRadians > PI) {
+	  objectArea = objectArea + areaOfExtraTriangle; 
+        } else {
+	  objectArea = objectArea - areaOfExtraTriangle;
+	}
+      }
+      else console.log(arguments[6]);
```
Comment (lm-n) [inline dist/p5-accessibility.js]: is this console log necessary?

File: src/entities/shapeEntity.js [hunk 3]
```diff
@@ -23,9 +23,31 @@ function ShapeEntity(Interceptor,shapeObject,arguments, canvasX, canvasY) {
...
+        let Cx = arguments[0] + (arguments[2]/2) * cos(arguments[5]).toFixed(2);
+        let Cy = arguments[1] + (arguments[3]/2) * sin(arguments[5]).toFixed(2);
+        let areaOfExtraTriangle = abs(Ax*(By - Cy) + Bx*(Cy - Ay) + Cx*(Ay - By))/2;
+        if(arcSizeInRadians > PI) {
+	  objectArea = objectArea + areaOfExtraTriangle; 
+        } else {
+	  objectArea = objectArea - areaOfExtraTriangle;
+	}
+      }
+      else console.log(arguments[6]);
```
Comment (lm-n) [inline src/entities/shapeEntity.js]: Is this console.log necessary?

File: src/entities/shapeEntity.js [hunk 4]
```diff
@@ -23,9 +23,31 @@ function ShapeEntity(Interceptor,shapeObject,arguments, canvasX, canvasY) {
...
+        let Bx = arguments[0] + (arguments[2]/2) * cos(arguments[4]).toFixed(2);
+        let By = arguments[1] + (arguments[3]/2) * sin(arguments[4]).toFixed(2);
+        let Cx = arguments[0] + (arguments[2]/2) * cos(arguments[5]).toFixed(2);
+        let Cy = arguments[1] + (arguments[3]/2) * sin(arguments[5]).toFixed(2);
+        let areaOfExtraTriangle = abs(Ax*(By - Cy) + Bx*(Cy - Ay) + Cx*(Ay - By))/2;
+        if(arcSizeInRadians > PI) {
+	  objectArea = objectArea + areaOfExtraTriangle; 
+        } else {
+	  objectArea = objectArea - areaOfExtraTriangle;
+	}
```
Comment (lm-n) [inline src/entities/shapeEntity.js]: fix indentation

File: dist/p5-accessibility.js [hunk 5]
```diff
@@ -1292,9 +1291,31 @@ Registry.register(FillEntity);
...
+        let Bx = arguments[0] + (arguments[2]/2) * cos(arguments[4]).toFixed(2);
+        let By = arguments[1] + (arguments[3]/2) * sin(arguments[4]).toFixed(2);
+        let Cx = arguments[0] + (arguments[2]/2) * cos(arguments[5]).toFixed(2);
+        let Cy = arguments[1] + (arguments[3]/2) * sin(arguments[5]).toFixed(2);
+        let areaOfExtraTriangle = abs(Ax*(By - Cy) + Bx*(Cy - Ay) + Cx*(Ay - By))/2;
+        if(arcSizeInRadians > PI) {
+	  objectArea = objectArea + areaOfExtraTriangle; 
+        } else {
+	  objectArea = objectArea - areaOfExtraTriangle;
+	}
```
Comment (lm-n) [inline dist/p5-accessibility.js]: fix indentation

**Comment by lm-n** [comment]
@mithru I've tested the code and it works! I did make some notes and requested some small changes. Also as you mentioned in #33 it would be good to include the type of arc i.e.: pie and maybe number of radians so that the output is something like this: 'orange(255,126,0) 100 degrees pie arc at top left covering 1.23% of the canvas'

**Comment by MathuraMG** [comment]
@mithru @lm-n  Sounds good, and looks like this is something we can pull out of the arguments and add to the name of the element. So double yes!

**Comment by mithru** [comment]
@lm-n hopefully it's all fine now? Sorry about the weirdness with the other push. forgot to branch :D

**Comment by lm-n** [comment]
@mithru not a problem! just make sure you create a new branch for each PR/issue

**Pull request merged**